src/classes/renderer/cache/png/persistent/engine.vala

```diff
@@ -0,0 +1,132 @@
+/**
+ * PNG cache store
+ *
+ * This file is part of pdfpc.
+ *
+ * Copyright (C) 2010-2011 Jakob Westhoff <jakob@westhoffswelt.de>
+ * Copyright 2012 David Vilar
+ * Copyright 2015 Andreas Bilke
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+namespace pdfpc.Renderer.Cache {
+    /**
+     * Cache store which holds all given items in memory as compressed png
+     * images and persists the cache on disk
+     */
+    public class PNG.Persistent.Engine: Renderer.Cache.PNG.Engine {
+        private static int cache_instance_counter = 0;
+
+        private int cache_instance_id;
+        private int? cache_width = null;
+        private int? cache_height = null;
+
+        protected string cache_directory;
+
+        /**
+         * Helper method for cache prepopulation, if the same caches are to be
+         * populated with different resolutions.
+         */
+        public static void reset_instance_counter() {
+            cache_instance_counter = 0;
+        }
+
+        public Engine( Metadata.Base metadata ) {
+            base( metadata );
+
+            cache_instance_counter++;
+            cache_instance_id = cache_instance_counter;
+
+            cache_directory = GLib.Environ.get_variable(GLib.Environ.get(), "HOME") + "/.cache/pdfpc/";
```

Comment:
Use `$XDG_CACHE_HOME/pdfpc` instead of hard coded path. If this variable does not exist, use `$HOME/.cache/pdfpc/` like in the XDG base directory specification